### ISSUE TITLE
Workflow with changePublishedState force_unpublished will create new version instead of unpublishing the object

### ISSUE THREAD

**Issue by mugge6** [pimcore/pimcore#7586]
# Bug Report

### Expected behavior

Workflow with two simple transitions (publish and unpublish) on DataObject:
```yaml
type: 'state_machine'
    marking_store:
        type: 'single_state'
        arguments: [ workflowState ]
```
...

```yaml
into_inactive:
into_unpublished:
    guard: is_fully_authenticated()
    from: [ published ]
    to: unpublished
    options:
        label: 'Unpublish'
        notes:
            commentEnabled: false
            commentRequired: false
        changePublishedState: force_unpublished
into_published:
    guard: is_fully_authenticated()
    from: [ unpublished ]
    to: publish
    options:
        label: 'Publish'
        notes:
            commentEnabled: false
            commentRequired: false
        changePublishedState: force_published
```
These transitions should toggle between publish and unpublish the DataObject.

### Actual behavior

Publishing is working fine. But unpublishing the DataObject will create a new version and leaves the object published. I expect, that the DataObject is unpublished afterwards.

**Comment by mugge6** [comment]
Looks like the method `applyWithAdditionalData` in Pimcore\Workflow\Manager is causing the issue.

```php
public function applyWithAdditionalData(Workflow $workflow, $subject, string $transition, array $additionalData, $saveSubject = false)
    {
        $this->notesSubscriber->setAdditionalData($additionalData);
        $marking = $workflow->apply($subject, $transition);
        $this->notesSubscriber->setAdditionalData([]);
        $transition = $this->getTransitionByName($workflow->getName(), $transition);
        $changePublishedState = $transition instanceof Transition ? $transition->getChangePublishedState() : null;
        if ($saveSubject && $subject instanceof AbstractElement) {
            if (method_exists($subject, 'getPublished')
                && (!$subject->getPublished() || $changePublishedState === ChangePublishedStateSubscriber::SAVE_VERSION)) {
                $subject->saveVersion();
            } else {
                $subject->save();
            }
        }
        return $marking;
    }
```

As far as I can see it shouldn't check for status:

```php
            // save independent from status
            if ($changePublishedState === ChangePublishedStateSubscriber::SAVE_VERSION) {

```

**Comment by fabfischer** [comment]
We also came across this problem. Is there a statement or an opinion on this from the official side?

**Comment by dvesh3** [comment]
seems related https://github.com/pimcore/pimcore/pull/7696

**Comment by ergosarapu** [comment]
Facing the same problem with `force_unpublished` (`force_published` works as expected). Creating a simple publish/unpublish workflow is blocked by this issue.

**Comment by dvesh3** [comment]
Fixed by #9028
